File: lista2_py/26.hipotenusa.py
#26. Leia os 3 (três) lados de um triângulo e identifique sua hipotenusa e seus catetos.

import math

def main():
    #Entrada
    lado_1 = obter_numero_float('Informe o valor do 1º lado do triângulo: ')
    lado_2 = obter_numero_float('Informe o valor do 2º lado do triângulo: ')
    lado_3 = obter_numero_float('Informe o valor do 3º lado do triângulo: ')

    #Processamento
    hipotenusa, cateto1, cateto2 = calcular_hipotenusa_catetos(lado_1, lado_2, lado_3)

    #Saída
    print(f'\n>> O valor da hipotenusa do triângulo é: {hipotenusa}cm;\n O valor dos catetos é: {cateto1} e {cateto2} cm.')

def obter_numero_float(descricao):
    numero = float(input(descricao))
    return numero

def calcular_hipotenusa_catetos(lado1,lado2,lado3):
    quadrado_lado1 = lado1 ** 2
    quadrado_lado2 = lado2 ** 2
    quadrado_lado3 = lado3 ** 2


    if (quadrado_lado1 == (quadrado_lado2 + quadrado_lado3)):
        hipotenusa = lado1
        cateto1 = lado2
        cateto2 = lado3
        return hipotenusa, cateto1, cateto2

    elif (quadrado_lado2 == (quadrado_lado1 + quadrado_lado3)):
        hipotenusa = lado2
        cateto1 = lado1
        cateto2 = lado3
        return hipotenusa, cateto1, cateto2

    elif (quadrado_lado3 == (quadrado_lado1 + quadrado_lado2)):
        hipotenusa = lado3
        cateto1 = lado1
        cateto2 = lado2
        return hipotenusa, cateto1, cateto2
        
main()

File: lista2_js.js/26.hipotenusa.js
//26. Leia os 3 (três) lados de um triângulo e identifique sua hipotenusa e seus catetos.
import { question } from "readline-sync"

function main() {
    // Entrada
    const lado_1 = obterNumeroFloat('Informe o valor do 1º lado do triângulo: ')
    const lado_2 = obterNumeroFloat('Informe o valor do 2º lado do triângulo: ')
    const lado_3 = obterNumeroFloat('Informe o valor do 3º lado do triângulo: ')

    // Processamento
    const [hipotenusa, cateto1, cateto2] = calcularHipotenusaCatetos(lado_1, lado_2, lado_3)

    // Saída
    console.log(`\n>> O valor da hipotenusa do triângulo é: ${hipotenusa}cm;\n O valor dos catetos é: ${cateto1} e ${cateto2} cm.`);
}

function obterNumeroFloat(descricao) {
    const numero = Number(question(descricao))
    return numero
}

function calcularHipotenusaCatetos(lado1, lado2, lado3) {
    const quadradoLado1 = lado1 ** 2
    const quadradoLado2 = lado2 ** 2
    const quadradoLado3 = lado3 ** 2

    if (quadradoLado1 === quadradoLado2 + quadradoLado3) {
        const hipotenusa = lado1
        const cateto1 = lado2
        const cateto2 = lado3
        return [hipotenusa, cateto1, cateto2]
    } else if (quadradoLado2 === quadradoLado1 + quadradoLado3) {
        const hipotenusa = lado2
        const cateto1 = lado1
        const cateto2 = lado3
        return [hipotenusa, cateto1, cateto2]
    } else if (quadradoLado3 === quadradoLado1 + quadradoLado2) {
        const hipotenusa = lado3
        const cateto1 = lado1
        const cateto2 = lado2
        return [hipotenusa, cateto1, cateto2]
    }
}

main()
